update-dois/src/main/java/org/reactome/release/updateDOIs/NewDOIChecker.java

```diff
@@ -0,0 +1,138 @@
+package org.reactome.release.updateDOIs;
+
+import java.util.Collection;
+
+import org.gk.model.GKInstance;
+import org.gk.model.InstanceDisplayNameGenerator;
+import org.gk.model.PersistenceAdaptor;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.SchemaClass;
+import org.gk.schema.InvalidAttributeException;
+import org.gk.schema.InvalidAttributeValueException;
+import org.gk.util.GKApplicationUtilities;
+
+public class NewDOIChecker {
+
+  private MySQLAdaptor dbaTestReactome;
+  private MySQLAdaptor dbaGkCentral;
+
+  // Create adaptors for test_reactome and gk_central
+  public void setTestReactomeAdaptor(MySQLAdaptor adaptor) {
+    this.dbaTestReactome = adaptor;
+  }
+
+  public void setGkCentralAdaptor(MySQLAdaptor adaptor) {
+    this.dbaGkCentral = adaptor;
+  }
+
+  public void findNewDOIs(GKInstance instanceEdit) {
+    Collection<GKInstance> dois;
+    Collection<GKInstance> gkdois;
+    try {
+      dois = this.dbaTestReactome.fetchInstanceByAttribute("Pathway", "doi", "NOT REGEXP", "^10.3180");
+  // Line below is code used when testing this module; Allowed me to revert the doi attribute after modifying it. Make sure to uncomment other line below too.
+      // dois = this.dbaTestReactome.fetchInstanceByAttribute("Pathway", "DB_ID", "REGEXP", "8939211|9006115|9018678|9033241");
+
+      if (!dois.isEmpty()) {
+        for (GKInstance doi : dois) {
+
+          String stableIdFromDb = doi.getAttributeValue("stableIdentifier").toString();
+          // This way of acquiring stableId is probably not ideal.
+          String stableId[] = stableIdFromDb.split("] ");
+          String updatedDoi = "10.3180/" + stableId[1];
+      // Line below is again used for reverting DB changes.
+          // String updatedDoi = "needs DOI";
+          String dbId = doi.getAttributeValue("DB_ID").toString();
+
+          doi.setAttributeValue("doi", updatedDoi);
+          this.dbaTestReactome.updateInstanceAttribute(doi, "doi");
+
+          // Grabs instance from gk central based on DB_ID taken from test_reactome, and updated doi field
+          gkdois = this.dbaGkCentral.fetchInstanceByAttribute("Pathway", "DB_ID", "=", dbId);
+          if (!gkdois.isEmpty()) {
+            for (GKInstance gkdoi : gkdois) {
```

Comment:
Maybe there should be some message written to the application log here, stating that an update was performed? That way when it runs we'll know what it did.

Reply:
Logging now happens after both databases have been updated